patches/minecraft/net/minecraft/item/ItemStack.java.patch

```diff
@@ -284,4 +284,18 @@
 +            return this.capabilities.areCompatible(other.capabilities);
 +        }
 +    }
++
++    public static boolean areItemStacksWithShareTagEqual(@Nullable ItemStack stackA, @Nullable ItemStack stackB)
++    {
++        if (stackA == null && stackB == null) return true;
++        if (stackA == null && stackB != null) return false;
++        if (stackA != null && stackB == null) return false;
```

Comment:
Actually you're right, my brain must have frozen right there. But honestly your way seems to confuse me because it seems like you're comparing two instance references when in fact you just want to make sure that both are null. So I will keep those two lines separate so that the intent is more clear unless someone tells me I have to merge them.